test/Quantum_test.jl

```diff
@@ -152,5 +152,82 @@
                 (inds(mpo; at=i), inds(mpo; at=i')) ⊆ inds(tensors(mpo; at=i))
             end
         end
+
+        @testset "mps greater than mpo" begin
+            # MPS-like tensor network with 4 sites
+            mps4sites = Quantum(
+                TensorNetwork(
+                    Tensor[
+                        Tensor(rand(2, 2), [:i, :j]),
+                        Tensor(rand(2, 2, 2), [:j, :k, :l]),
+                        Tensor(rand(2, 2, 2), [:l, :m, :n]),
+                        Tensor(rand(2, 2), [:n, :o]),
+                    ],
+                ),
+                Dict(site"1" => :i, site"2" => :k, site"3" => :m, site"4" => :o),
+            )
+
+            # MPO-like tensor network with 3 sites
+            mpo3sites = Quantum(
+                TensorNetwork(
+                    Tensor[
+                        Tensor(rand(2, 2, 2), [:i, :j, :k]),
+                        Tensor(rand(2, 2, 2, 2), [:l, :m, :k, :n]),
+                        Tensor(rand(2, 2, 2), [:o, :p, :n]),
+                    ],
+                ),
+                Dict(site"1" => :i, site"1'" => :j, site"2" => :l, site"2'" => :m, site"3" => :o, site"3'" => :p),
+            )
+
+            Tenet.@reindex! outputs(mps4sites) => inputs(mpo3sites)
+
+            mps4inds = [inds(mps4sites; at=i) for i in sites(mps4sites; set=:outputs)]
+            mpo3inds = [inds(mpo3sites; at=i) for i in sites(mpo3sites; set=:inputs)]
+            # test MPO sites indices are a subset of MPS's but not the opposite
+            @test mpo3inds ⊆ mps4inds
+            @test !(mps4inds ⊆ mpo3inds)
```

Comment:
you should go site by site checking that the index is the same. here a potential bug where the indices are permuted would go unnoticed.

```suggestion
            for lane in lanes(mpo3sites)
                @test inds(mps4sites; at=Site(lane)) == inds(mpo3sites; at=Site(lane; dual=true))
            end
```